features/steps/steps.py

```diff
@@ -211,14 +239,59 @@ def instance_getter(i,representation_id, representation_type, negative=False):
         if condition(i, representation_id, representation_type):
             return i
 
+def strip_split(stmt, strp = ' ', splt = ' '):
+    return list(
+        map(str.lower, map(lambda s: s.strip(strp), stmt.split(splt)))
+    )
+
+def include_subtypes(stmt):
+    stmt = strip_split(stmt, strp = '[]')
+    if len(stmt) > 1 and 'subtypes' in stmt:
+        excluding_statements = ['without', 'not', 'excluding', 'no']
+        if len(set(stmt).intersection(set(excluding_statements))):
+            return False
+        else:
+            return True
+    else:
+        return True
+
+def map_state(values, fn):
+    if isinstance(values, (tuple, list)):
+        return type(values)(map_state(v, fn) for v in values)
+    else:
+        return fn(values)
+
+@given("An {entity_opt_stmt}")
+def step_impl(context, entity_opt_stmt):
+    entity = entity_opt_stmt.split()[0]
+
+    try:
+        context.instances = context.model.by_type(entity, include_subtypes = include_subtypes(entity_opt_stmt))
+    except:
+        context.instances = []
+
+@given("{the_or_all} instances of {entity_opt_stmt}")
+def step_impl(context, the_or_all, entity_opt_stmt):
+    the_or_all = the_or_all
+
+    entity = entity_opt_stmt.split()[0]
 
-@given("An {entity}")
-def step_impl(context, entity):
     try:
-        context.instances = context.model.by_type(entity)
+        context.instances = context.model.by_type(entity, include_subtypes = include_subtypes(entity_opt_stmt))
+        within_model = 'all' in the_or_all.lower()
     except:
         context.instances = []
 
+    context.within_model = getattr(context, 'within_model', True) and within_model
+
+@given('Its attribute {attribute}')
+def step_impl(context, attribute):
+    context._push()
+    context.instances = map_state(context.instances, lambda i: getattr(i, attribute, None))
+    setattr(context, 'instances', context.instances)
```

Comment:
As said earlier in another PR, don't use setattr in these cases where the attribute is a static string.

Reply:
Deleted it in GEM003, but not here. Deleted in https://github.com/buildingSMART/ifc-gherkin-rules/commit/4dcfb4c509b06a23275b5c5af3e031dc0dd20677

What do you think is a good workflow for overlapping pull requests? Merging them every time after implementing fixes? It seems too easy to miss these things when doing it manually.